wordpress-mobile/WordPressKit-iOS
# [Update Username] Map new property in account settings GET and new method

### Description
This is part of [#9705](https://github.com/wordpress-mobile/WordPress-iOS/issues/9705)
This PR add a new property `usernameCanBeChanged` to map when the `GET me/settings` is called.
This property is used to define if a user is able to change the username. 

There's also a new method added to validate the current username. It's a GET `me/username/validate/USERNAME`

### Testing Details
- Run the test `AccountSettingsRemoteTests`
- Run [#12198](https://github.com/wordpress-mobile/WordPress-iOS/pull/12198)
